jenkins/macros.j2

```diff
@@ -89,3 +89,33 @@
     }
   },
 {% endmacro %}
+
+{% macro upload_artifacts(tag, filenames, folders=[]) %}
```

Comment:
i think in Python we would avoid kwarg with an empty list initializer (e.g. `folders=None`), possible to do that here? i think that amounts to `for folder in (folders or [])`

Reply:
i think you missed this one, but i don't think it's abused anywhere in here. let's merge this to alleviate CI pain and can fix in a follow-on